File: LeetCodeDaily/24_appendTtwoArrays.py
# 3069. Distribute Elements Into Two Arrays I
def solve(nums):
    arr1 = [nums[0]]
    arr2 = [nums[1]]
    n = len(nums)
    for i in range(2,n):
        l1 = arr1[-1]
        l2 = arr2[-1]
        if l1 > l2 :
            arr1.append(nums[i])
        else :
            arr2.append(nums[i])
        

    return arr1 + arr2

def printList(nums):
    flag = True
    for num in nums :
        if flag :
            print(" ",end="")
        print(num,end="")

def main():
    n = int(input())
    nums = list(map(int,input().split()))
    ans = solve(nums)
    printList(ans)


main()

File: LeetCodeDaily/24_appendTtwoArrays.cpp
#include <bits/stdc++.h>
using namespace std;
// 3069. Distribute Elements Into Two Arrays I
vector<int> resultArray(vector<int> &nums)
{
    int n = nums.size();

    vector<int> arr1 = {nums[0]};
    vector<int> arr2 = {nums[1]};

    for (int i = 2; i < n; i++)
    {
        if (arr1.back() > arr2.back())
        {
            arr1.push_back(nums[i]);
        }
        else
        {
            arr2.push_back(nums[i]);
        }
    }

    arr1.insert(arr1.end(), arr2.begin(), arr2.end());

    return arr1;
}

void printArray(vector<int>& nums){
    bool flag = false;
    for(auto& num : nums){
        if(flag) cout << " ";
        cout << num;
        flag = true;
    }
}

int main()
{
    int n;
    cin >> n;
    vector<int> nums(n);
    for(auto& num : nums){
        cin >> num;
    }

    vector<int> ans = resultArray(nums);

    printArray(ans);
    return 0;
}